optimum/exporters/openvino/__main__.py

```diff
@@ -621,3 +599,187 @@ def maybe_convert_tokenizers(library_name: str, output: Path, model=None, prepro
                     export_tokenizer(tokenizer, output / tokenizer_name, task=task)
     else:
         logger.warning("Tokenizer won't be converted.")
+
+
+def _apply_generic_quantization(
+    model_name_or_path: str,
+    original_task: str,
+    task: str,
+    library_name: str,
+    quantization_config: Union[Dict, "OVQuantizationConfigBase"],  # noqa: F821
+    output: Path,
+    cache_dir: str,
+    trust_remote_code: bool = False,
+    model_kwargs: Optional[Dict[str, Any]] = None,
+):
+    """
+    Apply quantization with a general quantization config to the exported model and save it to disk.
+    """
+    from ...intel.openvino.utils import _HEAD_TO_AUTOMODELS, TemporaryDirectory
+    from ...intel.utils.import_utils import DIFFUSERS_IMPORT_ERROR, is_diffusers_available
+
+    # Step 1. Obtain the correct OpenVINO model class
+    if library_name == "diffusers":
+        if not is_diffusers_available():
+            raise ValueError(DIFFUSERS_IMPORT_ERROR.format("Export of diffusers models"))
+
+        from diffusers import DiffusionPipeline
+
+        diffusers_config = DiffusionPipeline.load_config(model_name_or_path)
+        class_name = diffusers_config.get("_class_name", None)
+        ov_class_name = f"OV{class_name}"
+        try:
+            model_cls = getattr(__import__("optimum.intel", fromlist=[ov_class_name]), ov_class_name)
+        except (AttributeError, ImportError) as e:
+            raise RuntimeError(f"Wasn't able to locate OpenVINO class for {class_name} diffusion model.") from e
+    else:
+        try:
+            model_cls_name = _HEAD_TO_AUTOMODELS[task.replace("-with-past", "")]
+            if model_cls_name == "OVModelForFeatureExtraction" and library_name == "sentence_transformers":
+                model_cls_name = "OVSentenceTransformer"
+            model_cls = getattr(__import__("optimum.intel", fromlist=[model_cls_name]), model_cls_name)
+        except (AttributeError, ImportError, KeyError) as e:
+            raise RuntimeError(f"Wasn't able to locate OpenVINO class for task {original_task} ({task}).") from e
+
+    # Step 2. Load the exported model
+    additional_model_kwargs = {"use_cache": task.endswith("with-past")} if "generation" in task else {}
+    model = model_cls.from_pretrained(
+        output,
+        compile=False,
+        trust_remote_code=trust_remote_code,
+        cache_dir=cache_dir,
+        **((model_kwargs or {}) | additional_model_kwargs),
+    )
+
+    # Step 3. Apply quantization
+    with TemporaryDirectory() as tmpdir:
+        # Save quantized model to a temporary directory to avoid conflicts when reading and writing from the same directory
+        model._apply_quantization(
```

Comment:
why not give the `quantization_config` when calling `from_pretrained` above instead ?